### PR TITLE
bpo-41490: Update ensurepip to install pip 20.2.1 and setuptools 49.2.1

### PR DESCRIPTION
**Pull request by zooba**
https://bugs.python.org/issue41490

### CODE REVIEW

**Review by ned-deily — APPROVED** [review]

**Comment by vstinner** [comment]
The Travis CI job was blocked, so I closed/reopened the PR to see if it is enough to be able to merge the PR.

**Comment by vstinner** [comment]
Good: Travis CI job ran this time.

Bad: this PR is affected by https://bugs.python.org/issue41762

**Comment by vstinner** [comment]
New close/reopen to re-run all CI jobs, to see if it magically solves https://bugs.python.org/issue41762

**Comment by vstinner** [comment]
Tests / Windows (x86)

```
FAIL: test_with_pip (test.test_venv.EnsurePipTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\a\cpython\cpython\lib\test\test_venv.py", line 476, in do_test_with_pip
    self.run_with_capture(venv.create, self.env_dir,
  File "D:\a\cpython\cpython\lib\test\test_venv.py", line 76, in run_with_capture
    func(*args, **kwargs)
subprocess.CalledProcessError: Command '['C:\\Users\\runneradmin\\AppData\\Local\\Temp\\tmp1ihg5wsq\\Scripts\\python.exe', '-Im', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1.

(...)

Traceback (most recent call last):
  File "D:\a\cpython\cpython\lib\shutil.py", line 613, in _rmtree_unsaf
    os.unlink(fullname)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpeu5ljzar\\pip-20.2.1-py2.py3-none-any.whl'
```

Other tests failed:

```
2 tests failed:
    test_importlib test_venv

4 re-run tests:
    test__xxsubinterpreters test_asyncio test_importlib test_venv
```

importlib failure:

```
0:06:59 load avg: 2.78 [245/423/2] test_bigmem passed -- running: test_largefile (3 min 27 sec), test_peg_generator (38.6 sec), test_io (3 min 11 sec)
Failed to import test module: test.test_importlib.test_resource
Traceback (most recent call last):
  File "D:\a\cpython\cpython\lib\unittest\loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "D:\a\cpython\cpython\lib\unittest\loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "D:\a\cpython\cpython\lib\test\test_importlib\test_resource.py", line 10, in <module>
    from test.support import import_helper, unlink
ImportError: cannot import name 'unlink' from 'test.support' (D:\a\cpython\cpython\lib\test\support\__init__.py)

test test_importlib crashed -- Traceback (most recent call last):
  File "D:\a\cpython\cpython\lib\test\libregrtest\runtest.py", line 272, in _runtest_inner
    refleak = _runtest_inner2(ns, test_name)
  File "D:\a\cpython\cpython\lib\test\libregrtest\runtest.py", line 236, in _runtest_inner2
    test_runner()
  File "D:\a\cpython\cpython\lib\test\libregrtest\runtest.py", line 210, in _test_module
    raise Exception("errors while loading tests")
0:07:00 load avg: 2.78 [246/423/3] test_importlib failed -- running: test_largefile (3 min 27 sec), test_peg_generator (39.2 sec), test_io (3 min 11 sec)
Exception: errors while loading tests
```

File: Lib/test/test_importlib/test_resource.py [hunk 1]
```diff
@@ -1,10 +1,13 @@
 import sys
 import unittest
+import uuid
 
 from . import data01
 from . import zipdata01, zipdata02
 from . import util
 from importlib import resources, import_module
+from pathlib import Path
+from test.support import import_helper, unlink
```
Comment (vstinner) [inline Lib/test/test_importlib/test_resource.py:10]: unlink should now be imported from test.support.os_helper.

**Comment by ambv** [comment]
Note: this should be merged ASAP since the "backports" were.

**Comment by vstinner** [comment]
New close/reopen attempt to fix the CI.

**Comment by vstinner** [comment]
> Note: this should be merged ASAP since the "backports" were.

I wrote PR #22281 to fix https://bugs.python.org/issue41762 but my current PR is not correct (breaks the rendered documentation).

**Comment by vstinner** [comment]
I fixed https://bugs.python.org/issue41762 Let me try to close/reopen the issue to see if it's enough to run CI tests on a rebased PR.

**Comment by vstinner** [comment]
@zooba: Since https://bugs.python.org/issue41762 is now fixed, can you please update Lib/test/test_importlib/test_resource.py? See my previous comment:

> unlink should now be imported from test.support.os_helper.

**Comment by pablogsal** [comment]
You probably want to rebase this on top of https://github.com/python/cpython/pull/22527 (an drop the pip version bump).

**Comment by zooba** [comment]
Starting a fresh PR - apologies for the noise.

**Pull request closed without merging**